diffsmhm/diff_stats/cpu/wprp.py

```diff
@@ -44,16 +44,20 @@ def wprp_serial_cpu(
         The gradients of the projected correlation function.
     """
 
+    # check for first rpbin being zero for consistency with mpi kernels
+    # even though we don't use the 0.0 in this kernel?
+    assert np.allclose(rpbins_squared[0], 0)
+
     n_grads = w1_jac.shape[0]
-    n_rp = rpbins_squared.shape[0] - 1
+    n_rp = rpbins_squared.shape[0] - 2
```

Comment:
So a comment in the code that the -2 is from the fact that rp_bins are the bin edges so it is +1 relative to the number of bins, and then we add zero so that is another +1, which results in us having to subtract 2, would be good to add.